service/service_construction.go

```diff
@@ -786,6 +844,85 @@ func (s ConstructionService) createUnwrapPayload(
 	return tx, unsignedTx, &checkFrom, nil
 }
 
+func (s ConstructionService) createGenericContractCallPayload(req *types.ConstructionPayloadsRequest) (*ethtypes.Transaction, *transaction, *string, *types.Error) {
+	operationDescriptions, err := s.CreateGenericContractCallOperationDescription(req.Operations)
+	if err != nil {
+		return nil, nil, nil, WrapError(ErrInvalidInput, err.Error())
+	}
+
+	descriptions := &parser.Descriptions{
+		OperationDescriptions: operationDescriptions,
+		ErrUnmatched:          true,
+	}
+
+	matches, err := parser.MatchOperations(descriptions, req.Operations)
+	if err != nil {
+		return nil, nil, nil, WrapError(ErrInvalidInput, "unclear intent")
+	}
+
+	fromOp, amount := matches[0].First()
```

Comment:
Can you add a check to make sure matches list indeed has two elements?

Reply:
sure